lib/src/bootloader.rs

```diff
@@ -26,3 +26,94 @@ pub(crate) fn install_via_bootupd(
         .verbose()
         .run()
 }
+
+#[context("Installing bootloader using zipl")]
+pub(crate) fn install_via_zipl(device: &Utf8Path, boot_uuid: &str) -> Result<()> {
+    // Identify the target boot partition from UUID
+    let fs = crate::mount::inspect_filesystem_by_uuid(boot_uuid)?;
+    let boot_dir = Utf8Path::new(&fs.target);
+    let maj_min = fs.maj_min;
+
+    // Ensure that the found partition is a part of the target device
+    crate::blockdev::list_dev(device)?
+        .children
+        .with_context(|| format!("no partition found on {device:?}"))?
+        .iter()
+        .find(|dev| dev.maj_min == Some(maj_min.clone()))
+        .with_context(|| format!("partition device {:?} is not on {:?}", maj_min, device))?;
+
+    // Get an offset of the target partition by reading /sys/dev/block/<major:minor>/start
+    let sysfs_start_path = Utf8Path::new("/sys/dev/block").join(&maj_min).join("start");
+    let boot_offset = match std::fs::read_to_string(sysfs_start_path.clone()) {
+        Ok(v) => v.trim().parse::<i64>().map_err(|e| anyhow!(e)),
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => Ok(0), // partition-less disk
+        Err(e) => Err(anyhow!(e)),
+    }
+    .with_context(|| format!("reading {sysfs_start_path:?}"))?;
```

Comment:
Also shouldn't need `:?` (here and a variety of other places)

Further explanatory info: Often in Rust codebases that are operating on `std::path::Path` which is *not* statically known to be UTF-8, you'll see use of `:?` because it forces conversion to a human-readable debug.

A big part of the value of using [camino](https://docs.rs/camino/latest/camino/) is in things like this - we can safely treat the pathnames as strings too where it's convenient.

Reply:
done